### PR TITLE
Make sqllogictest platform-independent for the sign of NaN

### LINKED ISSUE #7458: Mismatch in query results in the local machine
When I run the sqllogictests on my computer, all query results including a NaN value give an error in scalar.slt file. Some examples:
```
External error: query result mismatch:
[SQL] select round(ln(a), 5), round(ln(b), 5), round(ln(c), 5) from signed_integers;
[Diff] (-expected|+actual)
-   -NaN 4.60517 -NaN
-   -NaN 9.21034 -NaN
-   0.69315 -NaN 4.81218
-   1.38629 NULL NULL
+   0.69315 NaN 4.81218
+   1.38629 NULL NULL
+   NaN 4.60517 NaN
+   NaN 9.21034 NaN
at test_files/scalar.slt:543
```

```
External error: query result mismatch:
[SQL] select log(a, 64) a, log(b), log(10, b) from signed_integers;
[Diff] (-expected|+actual)
-   -NaN 2 2
-   -NaN 4 4
    3 NULL NULL
-   6 -NaN -NaN
+   6 NaN NaN
+   NaN 2 2
+   NaN 4 4
at test_files/scalar.slt:595
```

Rows with NaN values are somehow listed towards the end of the table. So the tests don't pass in my local but in CI I don't get an error. Do you have any opinion about this problem?

### PR DESCRIPTION
**Pull request by sarutak**
## Which issue does this PR close?

Closes #7458 

## Rationale for this change
The sign of `NaN` returned from a function seems to depend on platform.
But it's better for `sqllogictest` to be platform-independent.

## What changes are included in this PR?
Revert #7403 and change one test which is sign sensitive .

## Are these changes tested?

`cargo test --test sqllogictests` passed.

## Are there any user-facing changes?
No.

### CODE REVIEW

File: datafusion/sqllogictest/test_files/predicates.slt [hunk 1]
```diff
@@ -254,39 +254,45 @@ foo
...
-query R
-SELECT * FROM test_float WHERE column1 IN (0.0, -1.2, NULL)
+query T
+SELECT column1 FROM test_float WHERE column2 IN (0.0, -1.2, NULL)
 ----
 
-query R
-SELECT * FROM test_float WHERE column1 IN (0.0, 1.2, 'NaN'::double, '-NaN'::double)
+query T
+SELECT column1 FROM test_float WHERE column2 IN (0.0, 1.2, 'NaN'::double, '-NaN'::double)
```
Comment (sarutak) [inline datafusion/sqllogictest/test_files/predicates.slt:291]: Modified this test which is sign sensitive not to show `NaN` and `-NaN`.

File: datafusion/sqllogictest/src/engines/conversion.rs [hunk 2]
```diff
@@ -41,11 +41,7 @@ pub(crate) fn varchar_to_str(value: &str) -> String {
 
 pub(crate) fn f16_to_str(value: f16) -> String {
     if value.is_nan() {
-        if value.is_sign_positive() {
-            "NaN".to_string()
-        } else {
-            "-NaN".to_string()
-        }
+        "NaN".to_string()
```
Comment (viirya) [inline datafusion/sqllogictest/src/engines/conversion.rs:44]: Hmm, you mean on different platform, a same value could be positive and negative?

Comment (viirya) [reply datafusion/sqllogictest/src/engines/conversion.rs:44]: If so, maybe it good to leave a comment here.

Comment (sarutak) [reply datafusion/sqllogictest/src/engines/conversion.rs:44]: Yes. a function can returns `-NaN` on x86 while 'NaN` on Arm.

File: datafusion/sqllogictest/test_files/select.slt [hunk 3]
```diff
@@ -224,12 +224,6 @@ select
 ----
 false true false true true false false true false true true false true true false false true
 
-# select NaNs
-query RRRR
-select 'NaN'::double a, '-NaN'::double b, 'NaN'::float c, '-NaN'::float d
-----
-NaN -NaN NaN -NaN
```
Comment (viirya) [inline datafusion/sqllogictest/test_files/select.slt:231]: Even simply literals also are impacted, i.e. platform dependent?

Comment (sarutak) [reply datafusion/sqllogictest/test_files/select.slt:231]: I don't think so.
I believe `NaN` and `-NaN` are finally parsed [here](https://github.com/Alexhuszagh/rust-lexical/blob/main/lexical-parse-float/src/parse.rs#L929).
So, as long as `f32::NAN`, `f64::NAN`, `-f32::NAN` and `-f64::NAN` follows IEEE 754, simple literals should not be impacted.

**Comment by berkaysynnada** [comment]
Hi @sarutak.
Thank you for working on this issue. The tests pass now without any errors on my machine.

File: datafusion/sqllogictest/test_files/select.slt (hunk 3 above)
Comment (alamb) [reply datafusion/sqllogictest/test_files/select.slt:231]: I think we should leave the test to show the behavior of parsing `-NaN`.

**Review by alamb — APPROVED** [review]
Thanks @sarutak  -- I think this change is reasonable to me

File: datafusion/sqllogictest/test_files/select.slt (hunk 3 above)
Comment (sarutak) [reply datafusion/sqllogictest/test_files/select.slt]: @alamb 
`ToString::to_string` ignores the sign of NaNs so if the sign of NaNs should be printed in the test result, we need to do like [this](https://github.com/apache/arrow-datafusion/pull/7462/files#diff-4a5281b565761d7b0e1a53c80a44c3ef00d17855b418a39ddf4ceaef7230e46cL44).
But if we do so, the issue discussed in #7458 happens.
So, if we need to ensure a literal `"-NaN"::double` is parsed as `-NaN`, how about testing like [this](https://github.com/apache/arrow-datafusion/pull/7462/files#diff-aa021899ee5a6b4e3aa6812249dcf7e2fafcef0926ae287f6904188dffc60db2R229)?

**Comment by alamb** [comment]
Looks good to me -- thank you @sarutak

**Pull request merged**